src/libs/ajax/DatasetBuilder.ts

```diff
@@ -69,23 +95,88 @@ export interface GetConceptsResponse {
 type DatasetRequest = {
   cohorts: Cohort[];
   conceptSets: ConceptSet[];
-  valuesSets: { domain: string; values: DatasetBuilderValue[] }[];
+  valueSets: { domain: string; values: DatasetBuilderValue[] }[];
+};
+
+export type DatasetRequestAPI = {
+  cohorts: CohortAPI[];
+  conceptSets: ConceptSet[];
+  valueSets: { id: number; name: string; values: string[] }[];
 };
 
-type DatasetAccessRequest = {
+export type DatasetAccessRequest = {
   name: string;
   researchPurposeStatement: string;
   datasetRequest: DatasetRequest;
 };
 
+export type DatasetAccessRequestAPI = {
+  name: string;
+  researchPurposeStatement: string;
+  datasetRequest: DatasetRequestAPI;
+};
+
+export const convertValueSets = (valueSets: { domain: string; values: DatasetBuilderValue[] }[]) => {
+  return _.map(
+    (valueSet) => ({ id: 0, name: valueSet.domain, values: _.map(({ name }) => name, valueSets.values) }),
+    valueSets
+  );
+};
+
+export const convertCohorts = (cohorts: Cohort[]) => {
+  return _.map(
+    (cohort) => ({
+      name: cohort.name,
+      criteriaGroups: _.map(
+        (criteriaGroup) => ({
+          name: criteriaGroup.name,
+          mustMeet: criteriaGroup.mustMeet,
+          meetAll: criteriaGroup.meetAll,
+          count: criteriaGroup.count,
+          criteria: _.map((criteria) => convertCriteria(criteria), criteriaGroup.criteria),
+        }),
+        cohort.criteriaGroups
+      ),
+    }),
+    cohorts
+  );
+};
+
+export const convertCriteria = (criteria: AnyCriteria) => {
+  if (criteria.kind === 'range') {
+    return { kind: criteria.kind, name: criteria.name, id: criteria.id, low: criteria.low, high: criteria.high };
+  }
+  if (criteria.kind === 'list') {
+    return {
+      kind: criteria.kind,
+      name: criteria.name,
+      id: criteria.id,
+      values: _.map((value) => value.id, criteria.values),
+    };
+  }
+  return { kind: criteria.kind, name: criteria.name, id: criteria.id };
```

Comment:
there has to be a better way to do this! Something less redundant. I tried using a Map or a type with a flexible key ([key: string]: any)